drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -189,6 +189,9 @@ static int sdw_compute_group_params(struct sdw_bus *bus,
 	}
 
 	list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+		if (m_rt->stream->state < SDW_STREAM_PREPARED)
```

Comment:
I think the issue manifests because sdw_prepare_stream() looks a the m_rt associated with the stream but sdw_compute_group_params() looks at the bus->m_rt_list. Can you maybe filter this list in this function by passing the stream as an argument when called from sdw_prepare_stream()?

Reply:
I thought about that. sdw_compute_group_params() lools at the bus->m_rt_list is because it needs to calculate all the active stream's bandwidth in the bus. We assume that all m_rt in the bus are active, but that is not true. And we did similar change in https://github.com/thesofproject/linux/commit/d7722a643421. That's why I choose to set the PREPARE state earlier.
But I am fine with the passing `stream` solution, too. PR updated.